plugin/storage/grpc/factory.go

```diff
@@ -124,3 +124,8 @@ func (f *Factory) CreateArchiveSpanWriter() (spanstore.Writer, error) {
 	}
 	return f.archiveStore.ArchiveSpanWriter(), nil
 }
+
+// Close closes the resources held by the factory
```

Comment:
because we have this:
```
./plugin/storage/factory.go:278:			if closer, ok := factory.(io.Closer); ok {
```
I suggest adding this after struct declaration:
```
var _ io.Closer = (*Factory)(nil)
```